packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -219,17 +219,22 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     }
   }
 
+  bool _shouldThrow(int statusCode) {
+    return statusCode < 200 || statusCode > 299;
+  }
+
   RestResponse _formatRestResponse(Map<String, dynamic> res) {
-    try {
-      return RestResponse(data: res["data"] as Uint8List);
-    }
-    // This shouldn't happen.  RestResponse should be properly formatted from native
-    on Exception catch (e) {
-      throw ApiException(AmplifyExceptionMessages.missingExceptionMessage,
-          recoverySuggestion:
-              AmplifyExceptionMessages.missingRecoverySuggestion,
-          underlyingException: e.toString());
+    final statusCode = res['statusCode'] as int;
+    final headers = res['headers'] as Map?;
+    final response = RestResponse(
+      data: res["data"] as Uint8List?,
+      headers: headers == null ? null : Map<String, String>.from(headers),
+      statusCode: statusCode,
+    );
+    if (_shouldThrow(statusCode)) {
```

Comment:
I am not seeing a test for this case in the next file? Is it already tested by something not in diff I'm not seeing or am I otherwise missing it? If not, would it make sense to add a test for this case as it's a change in how/where we throw this.

All-around, nice job with this logic moving to dart! I think it's an improvement as it should be more consistent now.

Reply:
Good catch! Added tests. And thanks, I'm glad things are a bit more centralized now as well.